llms/mlx_lm/utils.py

```diff
@@ -188,5 +190,9 @@ def load(path_or_hf_repo: str) -> Tuple[nn.Module, PreTrainedTokenizer]:
     model.load_weights(list(weights.items()))
 
     mx.eval(model.parameters())
-    tokenizer = AutoTokenizer.from_pretrained(model_path)
+    if model.__class__ is plamo.Model:
+        trust_remote_code = resolve_trust_remote_code(None, config["tokenizer_class"], False, True)
```

Comment:
@mzbac what do you think about this as a stopgap for the qwen model also?

Reply:
My understanding is the new version of Transformer will automatically prompt the user if they try to load a custom tokenizer, so we don't have to manually call `resolve_trust_remote_code()`. Additionally, I am concerned that people may use the mlx-lm package not only as a CLI tool but also integrate it into their service. The terminal prompt may not be visible to them. Personally, I feel it would be better to provide an option in the load function where users can explicitly enable remote code and pass specific token configuration if they want.

Reply:
That's a good point re using it not as a CLI. So maybe we should go with a config option in the load function? That sounds pretty simple to me?

Reply:
Yeah, something like https://github.com/ml-explore/mlx-examples/blob/25404dd0008aafea2aebce68fafe70238ef2ff66/llms/mlx_lm/utils.py#L188-L211 but maybe using an explicit option as `tokenizer_config` instead of kwargs.

Reply:
Yea I think that's very reasonable. That will fix the use as an API and the new transformers will make it work nicely with the CLI.

Are you interested in adding it to your qwen PR and then we can rebase this one on that?

Reply:
Sure thing, I just double checked and the new version of transformers hasn't been released yet. We either have to wait until the release or use transformers's master branch in requirement.txt. Do you have any preference? @awni

Reply:
I think the release should be out any day now, they are due for a release, so let's plan to wait.

Reply:
(keeps it simple from a requirements perspective)